packages/flutter_tools/test/integration.shard/isolated/native_assets_test.dart

```diff
@@ -60,12 +60,61 @@ const String packageName = 'package_with_native_assets';
 
 const String exampleAppName = '${packageName}_example';
 
+String? _createIOSSimulator() {
```

Comment:
We have all this logic already in the device lab tests.

https://github.com/flutter/flutter/blob/e6fa8655819e9b92740911b49e8215efc6e52dab/dev/devicelab/bin/tasks/native_assets_ios_simulator.dart

> $ cd <...>/flutter/dev/devicelab && dart bin/test_runner.dart test -t native_assets_ios_simulator

Would you be able to check why the existing hot integration test was not failing on the ios simulator?

(I don't feel strongly opposed to adding the logic also here, but since we already have it in the device lab tests it feels a bit like unnecessary code duplication.)

Reply:
I think the problem is that when hot restart happens this is in the logs

```
Performing hot restart...
Restarted application in 325ms.

══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following ArgumentError was thrown building _FocusInheritedScope:
```

The test only checks for `Restarted application`, which is there, so it assumes it passed, the exception is printed after. I think this is handled bit differently in the integration tests that's why those catch the issue.

Reply:
But even if I detect the exception and return `TaskResult.failure` the task still exits with success for some reason.

Reply:
@dcharkes, this should fix the device lab test: https://github.com/flutter/flutter/pull/148752/commits/eb6b04cf39717460784123174778cb2142730cea

Should I remove the simulator from integration tests?

Reply:
I wouldn't mind having it there, it's easier to run locally than the devicelab test.

However, I believe we shouldn't be using infinite resources on CI. And the test coverage should be similar between the integration test with the simulator and the devicelab test with the simulator.

cc @jmagman @vashworth What's our policy on how much testing resources we use? And what's the policy on where simulator tests should run? (Integration suite vs device lab. I'm added the simulator to the device lab based on existing device lab tests that run the simulator.)

Reply:
The 4/4 shard already takes long time to complete (in fact it did timeout in previous run and I had to restart it). I think maybe it's a good idea to remove it.

Reply:
Yeah that test runs long already. We don't want to introduce flakes due to timeouts. Let's remove it from this PR.

That means that this PR will just be a test fix and a fix 👍